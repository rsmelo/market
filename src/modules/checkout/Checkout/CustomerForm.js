import React from 'react'
import { Field } from 'redux-form'

import InputField from '../../../components/InputField'
import normalizeCPF from '../../../utils/normalize/normalizeCPF'
import normalizePhone from '../../../utils/normalize/normalizePhone'

const maskCPF = '000.000.000-00'
const maskPhone = '(00) 0000-0000'


const CustomerForm = () => (
  <div>
    <h2>Dados Pessoais</h2>
    <label htmlFor="name">Nome</label>
    <Field
      id="name"
      name="name"
      type="input"
      component={InputField}
    />
    <label htmlFor="email">E-mail</label>
    <Field
      id="email"
      name="email"
      type="input"
      component={InputField}
    />
    <label htmlFor="personCode">CPF</label>
    <Field
      id="personCode"
      name="personCode"
      type="input"
      normalize={normalizeCPF}
      placeholder={maskCPF}
      component={InputField}
    />
    <label htmlFor="phone">Telefone</label>
    <Field
      id="phone"
      name="phone"
      type="input"
      normalize={normalizePhone}
      placeholder={maskPhone}
      component={InputField}
    />
  </div>
)

export default CustomerForm
