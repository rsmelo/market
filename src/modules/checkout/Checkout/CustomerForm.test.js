import React from 'react'
import renderer from 'react-test-renderer'

import CustomerForm from './CustomerForm'

jest.mock('redux-form', () => ({ Field: 'Field' }))

describe('AddressForm', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CustomerForm />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
