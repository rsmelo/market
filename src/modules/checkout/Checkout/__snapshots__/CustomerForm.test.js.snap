// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm renders correctly 1`] = `
<div
  className={undefined}
>
  <h2>
    1. Dados Pessoais
  </h2>
  <label
    htmlFor="name"
  >
    Nome
  </label>
  <Field
    component={[Function]}
    id="name"
    name="name"
    type="input"
  />
  <label
    htmlFor="email"
  >
    E-mail
  </label>
  <Field
    component={[Function]}
    id="email"
    name="email"
    type="input"
  />
  <label
    htmlFor="personCode"
  >
    CPF
  </label>
  <Field
    component={[Function]}
    id="personCode"
    name="personCode"
    normalize={[Function]}
    placeholder="000.000.000-00"
    type="input"
  />
  <label
    htmlFor="phone"
  >
    Telefone
  </label>
  <Field
    component={[Function]}
    id="phone"
    name="phone"
    normalize={[Function]}
    placeholder="(00) 0000-0000"
    type="input"
  />
</div>
`;
